### DISCUSSION TITLE
Q1 2024 Course Roadmap - New Lessons and Enhancements  🚀

### DISCUSSION THREAD

**Discussion by koreyspace** [microsoft/generative-ai-for-beginners#237] [Announcements]
**First off, I want to thank everyone who has contributed to this repo and course! It has been truly a community effort.** 

We have heard your feedback on the course through comments, issues and our community Discord. I want to present you our roadmap of for Q1 2024. 

|   Content   |  Description   |
|---|---|
|  Lesson 13: Securing Your Generative AI Applications  |   A lesson covering how to secure your GenAI apps from attacks like Prompt Inject as well as how to use tools in a secure way|
|  Lesson 14: The Generative AI Application Lifecycle  |  Lesson covering designing, testing, deploying and improving a Generative AI application  |
|  Lesson 15: Retrieval Augmented Generation and Vector Databases   |  Lesson covering different RAG techniques and how to integrate a Vector Database  |
|  Lesson 16: Building with Open Source Models    |  Lesson covering building GenAI applications with the available Open Source Models on tools such as HuggingFace   |
|  TypeScript Support    |  Creating versions of the code lessons to use TypeScript     |
|  OpenAI API Support    |  Creating versions of the code lessons to use the OpenAI API    |

If you have any questions or comments please feel free to post below!

**Comment by seguyy** [comment]
I wish people who don't work at a company at the moment would also have a chance to become eligible for Azure OpenAI Service. This unfortunately prevented me from starting the course.

**Reply by koreyspace** [reply]
Thanks for the feedback @seguyy - as mentioned in the roadmap we will be adding Open AI API support this quarter which will allow anyone to take this course if they have Open AI API access.

**Reply by ygrenzinger** [reply]
Eager to have access to the full course with Open AI API (not having access to Azure OpenAI service) :)

**Reply by koreyspace** [reply]
@ygrenzinger , we will be launching version 2 of the course on Feb 23r which will have OpenAI Support.

**Comment by Freshlystore** [comment]
I

**Reply by Icculusians** [reply]
I, AS WELL...! 

I

**Comment by Icculusians** [comment]
I

**Comment by NDT3117** [comment]
Thanks for sharing!

**Comment by DarshanRs1** [comment]
I am

**Comment by pseacrest** [comment]
This is a great share, thx

**Comment by Rad211** [comment]
Hello people I like what I see but I don't know where to start probably I will earn whit a little help 🤦
